[FEATURE] Ajout du taux de réussite d'un résultat thématique sur Pix-Admin (PIX-3507).

alexandrecoin opened the pull request:
## :jack_o_lantern: Problème

A l'heure actuelle, afin de créer les critères associés aux résultats thématiques, il faut passer par Adminer.

## :bat: Solution

Ajout d'un champ dans le formulaire de création de résultat thématique dans `pix-admin` avec la route API associée à cette nouvelle entrée.

## :spider_web: Remarques

Le dernier commit de renommage a été réalisé afin de correspondre davantage à l'implémentation actuelle, à savoir l'existence d'un seul critère de résultat thématique.

## :ghost: Pour tester

Se rendre sur Pix-admin
Profil Cible -> choisir un profil
Clés de lecture -> Nouveau résultat thématique

Lors de la création, vérifier que:
- Il est pas possible de créer un badge sans taux de réussite
- Si taux de réussite, le nombre doit être compris entre 0 et 100

alexandrecoin committed:
sr(admin): rename page title to meet business requirements

alexandrecoin committed:
feat(admin): create basic campaign participation threshold input

nlepage committed:
feat(api): add controller for badge-criteria creation

Commit:
feat(api): add repository to create badge-criteria

Commit:
fix(tests): add badgeId propertycin badgeCriteria model expectations

Commit:
feat(admin): add urlForCreateRecord in badge-criteria's adapter

Commit:
feat(admin): add logic for badge and badge-criterion creation

pix-service commented:
I'm deploying this PR to these urls:

- App (.fr): https://app-pr3689.review.pix.fr
- App (.org): https://app-pr3689.review.pix.org
- Orga: https://orga-pr3689.review.pix.fr
- Certif: https://certif-pr3689.review.pix.fr
- Admin: https://admin-pr3689.review.pix.fr
- API: https://api-pr3689.review.pix.fr/api/

 Please check it out!

alexandrecoin committed:
sr(api/admin): rename file and functions

As of the time of this PR, only the threshold criterion has been added,
hence the renaming to badge-criterion.

VincentHardouin commented on admin/app/components/target-profiles/badge-form.js:
```diff
@@ -16,22 +16,53 @@ export default class BadgeForm extends Component {
     } catch (error) {
       console.error(error);
-      this.notifications.error('Erreur lors de la création du badge.');
+      this.notifications.error('Erreur lors de la création du résultat thématique.');
```
Je descendrai ce `try/catch` dans la méthode `_createBadge`

alexandrecoin committed:
feat(admin): improve logic handling for badge creation

sbedeau reviewed (approved):
Ça m'a l'air bien.

Fonctionnellement (hors scope ?), j'ai remarqué qu'il faut reload la page à la première arrivée sur le tableau des RT pour les voir. Il faudra qu'on améliore ça

alexandrecoin commented:
> Fonctionnellement (hors scope ?), j'ai remarqué qu'il faut reload la page à la première arrivée sur le tableau des RT pour les voir. Il faudra qu'on améliore ça

Bien vu. Sans être forcément hors-scope, je pense toutefois que la PR est déjà assez grosse et que ce souci peut-être réglé dans un nouveau ticket.
